pkg/i2gw/providers/kong/converter.go

```diff
@@ -43,24 +45,37 @@ func newConverter() *converter {
 	}
 }
 
-func (c *converter) convert(storage *storage) (i2gw.GatewayResources, field.ErrorList) {
+func (c *converter) convert(storage storage) (i2gw.GatewayResources, field.ErrorList) {
```

Comment:
nit: I think when you call convert you do pass a pointer to storage

Reply:
nope, I don't

Reply:
any reason not to? We are copying the whole storage struct if we pass by value